/* Copyright 2020 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

 syntax = "proto3";
 import "tensorflow/core/framework/tensor.proto";
 import "fedlearner/common/common.proto";
 
 package fedlearner.common;
 
 service Scheduler {
     rpc ModelAuth(ModelAuthRequest) returns (Status);
     rpc SubmitTrainJob(TrainJobRequest) returns (Status);
 }
   
 message ModelVersion {
     string version = 1;
     string code_url = 2;
 }
 message ModelMeta {
     string model_uri  = 1;
     string image_uri = 3;
     string ps_command = 4;
     string worker_command = 5;
     string master_command = 6;
 }
 
 message ModelAuthRequest {
     ModelMeta model_meta = 1;
     ModelVersion model_version = 2;
 }
   
 message DataMeta {
   string data_source_name = 1;
   int64 start_time = 2;
   int64 end_time = 3;
 }
   
 message TrainJobRequest {
     string name = 1;
     string application_id = 2;
     ModelMeta model_meta = 3;
     ModelVersion model_version = 4;
     int32 pair_num = 5;
     DataMeta data_meta = 6;
     string description = 7;
 }